### DISCUSSION TITLE
Bluetooth Mesh working with legacy and extended advertising simultaneously

### DISCUSSION THREAD

**Discussion by venngar** [zephyrproject-rtos/zephyr#44315] [Q&A]
Hello everyone,

I've been working on a custom side project related to Bluetooth Mesh and Direction Finding in the same device (nRF5340, using nRF Connect SDK).

I'm aware that according to the documentation, for Bluetooth Mesh the advertising mode should be [either legacy or extended](https://docs.zephyrproject.org/latest/kconfig.html#CONFIG_BT_MESH_ADV) and that if I set CONFIG_BT_EXT_ADV=y, it means legacy advertisements for Bluetooth Mesh won't work. My goal though is to make it compatible so it can work with both.

So I'd like to know, is it possible theoretically to make Bluetooth Mesh support both advertisement types simultaneously? If yes, what are the steps to implement it if the source code of RTOS has to be changed? As far as I understand, Mesh extended advertisement support is enabled in Zephyr RTOS.

**Comment by carlescufi** [comment]
@jhedberg @PavelVPV @trond-snekvik

**Comment by PavelVPV** [comment]
Hi @venngar, I think mesh will still work normally if you enable the extended advertiser. You can check it with this sample for nRF Connect SDK: http://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/samples/bluetooth/mesh/ble_peripheral_lbs_coex/README.html.
